File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/playback/PlaybackState.java
package io.casey.musikcube.remote.playback;

public enum PlaybackState {
    Stopped("stopped"),
    Buffering("buffering"), /* streaming only */
    Playing("playing"),
    Paused("paused");

    private final String rawValue;

    PlaybackState(String rawValue) {
        this.rawValue = rawValue;
    }

    @Override
    public String toString() {
        return rawValue;
    }

    static PlaybackState from(final String rawValue) {
        if (Stopped.rawValue.equals(rawValue) || "unknown".equals(rawValue)) {
            return Stopped;
        }
        else if (Playing.rawValue.equals(rawValue)) {
            return Playing;
        }
        else if (Paused.rawValue.equals(rawValue)) {
            return Paused;
        }

        throw new IllegalArgumentException("rawValue is invalid");
    }
}

File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/service/playback/PlaybackState.kt
package io.casey.musikcube.remote.service.playback

enum class PlaybackState constructor(private val rawValue: String) {
    Stopped("stopped"),
    Buffering("buffering"), /* streaming only */
    Playing("playing"),
    Paused("paused");

    override fun toString(): String = rawValue

    companion object {
        internal fun from(rawValue: String): PlaybackState {
            if (Stopped.rawValue == rawValue || "unknown" == rawValue) {
                return Stopped
            }
            else if (Playing.rawValue == rawValue) {
                return Playing
            }
            else if (Paused.rawValue == rawValue || "prepared" == rawValue) {
                return Paused
            }

            throw IllegalArgumentException("rawValue '$rawValue' is unknown")
        }
    }
}
